This is the linking failure in the '[WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004](https://build.webkit.org/#/builders/2)' bot:

https://build.webkit.org/#/builders/2/builds/8866

```bash
FAILED: lib/libWPEPlatform-2.0.so.0.0.0 
: && /usr/lib/ccache/clang++-16 -fPIC -fdiagnostics-color=always -fcolor-diagnostics -Wextra -Wall -pipe -Wno-noexcept-type -Wno-psabi -Wno-misleading-indentation -Wno-parentheses-equality -Qunused-arguments -Wundef -Wpointer-arith -Wmissing-format-attribute -Wformat-security -Wcast-align -Wno-tautological-compare -fno-strict-aliasing -fno-exceptions -fno-rtti -ffunction-sections -fdata-sections -O3 -DNDEBUG  -Wl,--no-undefined -L/home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib -Wl,--gc-sections -Wl,--disable-new-dtags -shared -Wl,-soname,libWPEPlatform-2.0.so.0 -o lib/libWPEPlatform-2.0.so.0.0.0 Source/WTF/wtf/CMakeFiles/WTF.dir/ASCIICType.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ApproximateTime.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Assertions.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/AutomaticThread.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/BitVector.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/BloomFilter.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CPUTime.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ClockType.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CodePtr.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CompilationThread.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ConcurrentBuffer.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ConcurrentPtrHashSet.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CountingLock.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CrossThreadCopier.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CrossThreadTaskHandler.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CryptographicUtilities.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CryptographicallyRandomNumber.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/CurrentTime.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/DataLog.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/DateMath.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/DebugHeap.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/EmbeddedFixedVector.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/FastBitVector.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/FastFloat.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/FastMalloc.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/FilePrintStream.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/FileSystem.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/FunctionDispatcher.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Gigacage.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/GlobalVersion.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/GregorianDateTime.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/HashTable.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/HexNumber.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Int128.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/JSONValues.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Language.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/LikelyDenseUnsignedIntegerSet.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Lock.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/LockedPrintStream.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/LogChannels.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/LogInitialization.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Logger.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Logging.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/MainThread.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/MainThreadDispatcher.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/MediaTime.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/MemoryPressureHandler.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/MetaAllocator.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/MonotonicTime.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/NativePromise.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/NumberOfCores.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/OSRandomSource.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ObjectIdentifier.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/PageBlock.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ParallelHelperPool.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ParallelJobsGeneric.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ParkingLot.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/PrintStream.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ProcessPrivilege.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/RAMSize.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/RandomDevice.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ReadWriteLock.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/RefCounted.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/RefCountedLeakCounter.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/RunLoop.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/SHA1.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/SafeStrerror.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Seconds.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/SegmentedVector.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/SixCharacterHash.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/SmallSet.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/StackBounds.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/StackCheck.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/StackPointer.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/StackStats.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/StackTrace.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/StringPrintStream.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/SuspendableWorkQueue.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ThreadGroup.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/ThreadMessage.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Threading.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/TimeWithDynamicClockType.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/TimingScope.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/URL.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/URLHelpers.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/URLParser.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/UUID.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/UniqueArray.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/Vector.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WTFAssertions.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WTFConfig.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WTFProcess.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WallTime.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WeakPtr.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WeakRandomNumber.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WordLock.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WorkQueue.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/WorkerPool.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/dragonbox/dragonbox_to_chars.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/bignum-dtoa.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/bignum.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/cached-powers.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/diy-fp.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/double-conversion.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/fast-dtoa.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/fixed-dtoa.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/dtoa/strtod.cc.o Source/WTF/wtf/CMakeFiles/WTF.dir/persistence/PersistentCoders.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/persistence/PersistentDecoder.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/persistence/PersistentEncoder.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/ASCIILiteral.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/AtomString.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/AtomStringImpl.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/AtomStringTable.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/Base64.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/CString.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/ExternalStringImpl.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/LineEnding.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/StringBuffer.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/StringBuilder.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/StringBuilderJSON.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/StringCommon.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/StringImpl.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/StringView.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/SymbolImpl.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/SymbolRegistry.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/TextBreakIterator.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/TextStream.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/WTFString.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/icu/UTextProvider.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/icu/UTextProviderLatin1.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/icu/UTextProviderUTF16.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/threads/BinarySemaphore.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/threads/Signals.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unicode/CollatorDefault.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unicode/UTF8Conversion.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unicode/icu/CollatorICU.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unicode/icu/ICUHelpers.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/generic/MainThreadGeneric.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/generic/MemoryFootprintGeneric.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/generic/WorkQueueGeneric.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/Application.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/ChassisType.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/FileSystemGlib.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/GRefPtr.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/GSocketMonitor.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/RunLoopGLib.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/Sandbox.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/SocketConnection.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/glib/URLGLib.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/linux/CurrentProcessMemoryStatus.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/linux/RealTimeThreads.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/posix/CPUTimePOSIX.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/posix/FileSystemPOSIX.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/posix/OSAllocatorPOSIX.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/posix/ThreadingPOSIX.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/text/unix/TextBreakIteratorInternalICUUnix.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unix/LanguageUnix.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unix/LoggingUnix.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unix/MemoryPressureHandlerUnix.cpp.o Source/WTF/wtf/CMakeFiles/WTF.dir/unix/UniStdExtrasUnix.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/AllIsoHeaps.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Allocator.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/AvailableMemory.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Cache.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/CryptoRandom.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Deallocator.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/DebugHeap.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Environment.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/FreeList.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Gigacage.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Heap.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/HeapConstants.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/HeapKind.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoHeap.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoHeapImpl.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoMallocFallback.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoPage.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoSharedHeap.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoSharedPage.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoTLS.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoTLSEntry.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/IsoTLSLayout.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/LargeMap.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Logging.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Mutex.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/ObjectType.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/ObjectTypeTable.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/PerProcess.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/Scavenger.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/TZoneHeap.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/TZoneHeapManager.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/bmalloc/bmalloc.cpp.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/bmalloc_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/bmalloc_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/bmalloc_type.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/hotbit_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/hotbit_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/iso_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/iso_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/iso_test_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/iso_test_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/jit_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/jit_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/minalign32_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/minalign32_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pagesize64k_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pagesize64k_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_alignment.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_all_heaps.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_allocation_callbacks.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_allocation_result.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_all_shared_page_directories.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_baseline_allocator.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_baseline_allocator_table.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_basic_heap_config_enumerator_data.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bitfit_allocator.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bitfit_directory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bitfit_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bitfit_page.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bitfit_page_config_kind.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bitfit_size_class.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bitfit_view.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bootstrap_free_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_bootstrap_heap_page_provider.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_coalign.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_commit_span.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_committed_pages_vector.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_compact_bootstrap_free_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_compact_expendable_memory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_compact_heap_reservation.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_compact_large_utility_free_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_compute_summary_object_callbacks.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_create_basic_heap_page_caches_with_reserved_memory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_deallocate.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_debug_spectrum.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_deferred_decommit_log.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_designated_intrinsic_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_dyld_state.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_dynamic_primitive_heap_map.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_ensure_heap_forced_into_reserved_memory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_ensure_heap_with_page_caches.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerable_page_malloc.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerable_range_list.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerate_bitfit_heaps.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerate_initially_unaccounted_pages.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerate_large_heaps.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerate_segregated_heaps.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerate_unaccounted_pages_as_meta.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerator.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_enumerator_region.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_epoch.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_exclusive_view_template_memo_table.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_expendable_memory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_extended_gcd.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_fast_large_free_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_fast_megapage_cache.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_fast_megapage_table.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_fd_stream.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_free_granules.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_config_kind.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_config_utils.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_for_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_lock.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_ref.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_runtime_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_summary.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_heap_table.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_immortal_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_expendable_memory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_free_heap_deferred_commit_log.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_free_heap_helpers.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_heap_physical_page_sharing_cache.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_map.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_sharing_pool.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_large_utility_free_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_lenient_compact_unsigned_ptr.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_local_allocator.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_local_allocator_scavenger_data.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_local_view_cache.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_lock.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_lock_free_read_ptr_ptr_hashtable.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_log.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_malloc_stack_logging.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_medium_megapage_cache.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_megapage_cache.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_monotonic_time.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_page_base.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_page_base_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_page_header_table.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_page_malloc.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_page_sharing_participant.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_page_sharing_pool.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_payload_reservation_page_list.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_physical_memory_transaction.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_primitive_heap_ref.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_probabilistic_guard_malloc_allocator.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_ptr_worklist.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_race_test_hooks.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_random.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_red_black_tree.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_redundant_local_allocator_node.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_report_crash.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_reserved_memory_provider.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_root.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_scavenger.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_directory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_exclusive_view.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_page.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_page_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_page_config_kind.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_page_config_kind_and_role.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_partial_view.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_shared_handle.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_shared_page_directory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_shared_view.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_size_directory.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_segregated_view.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_shared_page_directory_by_size.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_simple_free_heap_helpers.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_simple_large_free_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_simple_type.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_status_reporter.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_stream.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_string_stream.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_thread_suspend_lock.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_thread_local_cache.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_thread_local_cache_layout.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_thread_local_cache_layout_node.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_thread_local_cache_node.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_utility_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_utility_heap_config.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_utils.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_versioned_field.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/pas_virtual_range.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/thingy_heap.c.o Source/bmalloc/CMakeFiles/bmalloc.dir/libpas/src/libpas/thingy_heap_config.c.o Source/WebKit/WPEPlatform/wpe/headless/CMakeFiles/WPEPlatformHeadless.dir/WPEDisplayHeadless.cpp.o Source/WebKit/WPEPlatform/wpe/headless/CMakeFiles/WPEPlatformHeadless.dir/WPEToplevelHeadless.cpp.o Source/WebKit/WPEPlatform/wpe/headless/CMakeFiles/WPEPlatformHeadless.dir/WPEViewHeadless.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEDisplayWayland.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEInputMethodContextWaylandV1.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEInputMethodContextWaylandV3.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEMonitorWayland.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEToplevelWayland.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEViewWayland.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEWaylandCursor.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEWaylandCursorTheme.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEWaylandSHMPool.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEWaylandSeat.cpp.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/__/__/__/__/__/DerivedSources/WPEPlatform/wpe/wayland/linux-dmabuf-unstable-v1-protocol.c.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/__/__/__/__/__/DerivedSources/WPEPlatform/wpe/wayland/text-input-unstable-v1-protocol.c.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/__/__/__/__/__/DerivedSources/WPEPlatform/wpe/wayland/text-input-unstable-v3-protocol.c.o Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/__/__/__/__/__/DerivedSources/WPEPlatform/wpe/wayland/xdg-shell-protocol.c.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEEvent.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEBuffer.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEBufferDMABuf.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEBufferDMABufFormats.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEBufferSHM.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEColor.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPECursorTheme.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEDisplay.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEEGLError.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEExtensions.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEInputMethodContext.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEInputMethodContextNone.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEKeyUnicode.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEKeymap.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEKeymapXKB.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEMonitor.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPERectangle.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEToplevel.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEVersion.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/wpe/WPEView.cpp.o Source/WebKit/WPEPlatform/CMakeFiles/WPEPlatform-2.0.dir/__/__/__/DerivedSources/WPEPlatform/wpe/WPEEnumTypes.cpp.o  /usr/lib/x86_64-linux-gnu/libepoxy.so  /home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib/libgio-2.0.so  /home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib/libgobject-2.0.so  /home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib/libglib-2.0.so  -lxkbcommon  /usr/lib/x86_64-linux-gnu/libgbm.so  /usr/lib/x86_64-linux-gnu/libepoxy.so  /usr/lib/x86_64-linux-gnu/libicudata.so  /usr/lib/x86_64-linux-gnu/libicui18n.so  /usr/lib/x86_64-linux-gnu/libicuuc.so  /usr/lib/x86_64-linux-gnu/libz.so  /lib/x86_64-linux-gnu/libsystemd.so  lib/libSysProfCapture.a  -lpthread  /usr/lib/x86_64-linux-gnu/libgbm.so  /home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib/libdrm.so  -ldl  -latomic  -lwayland-server  -lwayland-egl  -lwayland-client  /home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib/libgio-2.0.so  /home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib/libgobject-2.0.so  /home/buildbot/wpewebkit/nightly/WPE-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004/build/WebKitBuild/DependenciesWPE/Root/lib/libglib-2.0.so  -lxkbcommon && :
/usr/bin/ld: Source/WebKit/WPEPlatform/wpe/wayland/CMakeFiles/WPEPlatformWayland.dir/WPEWaylandSHMPool.cpp.o: in function `WPE::WaylandSHMPool::create(wl_shm*, unsigned long)':
WPEWaylandSHMPool.cpp:(.text._ZN3WPE14WaylandSHMPool6createEP6wl_shmm+0x1fe): undefined reference to `shm_open'
/usr/bin/ld: WPEWaylandSHMPool.cpp:(.text._ZN3WPE14WaylandSHMPool6createEP6wl_shmm+0x267): undefined reference to `shm_unlink'
```

--- a/Source/WebKit/WPEPlatform/wpe/wayland/CMakeLists.txt
+++ b/Source/WebKit/WPEPlatform/wpe/wayland/CMakeLists.txt
@@ -44,6 +44,14 @@ set(WPEPlatformWayland_LIBRARIES
     ${WPEPlatform_LIBRARIES}
 )
 
+check_function_exists(shm_open SHM_OPEN_EXISTS)
+if (NOT SHM_OPEN_EXISTS)
+    check_function_exists(shm_open SHM_OPEN_REQUIRES_LIBRT)
+    if (SHM_OPEN_REQUIRES_LIBRT)
+        list(APPEND WPEPlatformWayland_LIBRARIES rt)
+    endif ()
+endif ()
+
 set(WPEPlatformWayland_SOURCES_FOR_INTROSPECTION
     ${WPEPlatformWayland_INSTALLED_HEADERS}
     ${WPEPlatformWayland_SOURCES}

